rfcs/text/0006_management_section_service.md

```diff
@@ -0,0 +1,341 @@
+- Simple API that is designed in keeping with new platform principles
+  - This includes being rendering-framework-agnostic... You should be able to build your management section UI however you'd like
+- Clear separation of app/UI code and service code, even if both live within the same plugin
+- Flexibility to potentially support alternate layouts in the future (see mockups in [reference section](#reference) below)
+
+# Basic example
+This API is influenced heavily by the [application service mounting RFC](https://github.com/elastic/kibana/blob/master/rfcs/text/0004_application_service_mounting.md). The intent is to make the experience consistent with that service; the Management section is basically one big app with a bunch of registered "subapps".
+
+```ts
+// my_plugin/public/plugin.ts
+
+export class MyPlugin {
+  setup(core, { management }) {
+    // Registering a new link to a new section
+    const mySection = management.sections.register({
+      id: 'my-section',
+      title: 'My Main Section', // display name
+      description: 'hello', // not used in current UI, but possibly in future
+      order: 10,
+      euiIconType: 'iconName',
+    });
+    mySection.registerLink({
+      id: 'my-link',
+      title: 'My Link', // display name
+      description: 'hello', // not used in current UI, but possibly in future
+      order: 20,
+      async mount(context, params) {
+        const { renderApp } = await import('./my-section');
+        return renderApp(context, params);
+      }
+    });
+
+    // Registering a new link to an existing section
+    const kibanaSection = management.sections.get('kibana');
+    kibanaSection.registerLink({ id: 'my-kibana-section-link', ... });
+  }
+
+  start(core, { management }) {
+    // access all registered sections, filtered based on capabilities
+    management.sections.getAvailable();
+    // automatically navigate to any section link by id
+    management.sections.navigateToSectionLink('my-kibana-section-link');
+  }
+}
+ 
+// my_plugin/public/my-section.tsx
+
+export function renderApp(context, { sectionBasePath, element }) {
+  ReactDOM.render(
+    // `sectionBasePath` would be `/app/management/my-section/my-link`
+    <MyApp basename={sectionBasePath} />,
+    element
+  );
+ 
+  // return value must be a function that unmounts (just like Core Application Service)
+  return () => ReactDOM.unmountComponentAtNode(element);
+}
+```
+
+We can also create a utility in `kibana_react` to make it easy for folks to `mount` a React app:
+```ts
+// src/plugins/kibana_react/public/mount_with_react.tsx
+import { KibanaContextProvider } from './context';
+
+export const mountWithReact = (
+  Component: React.ComponentType<{ basename: string }>,
+  context: AppMountContext,
+  params: ManagementSectionMountParams,
+) => {
+  ReactDOM.render(
+    (
+      <KibanaContextProvider services={{ ...context }}>
+        <Component basename={params.sectionBasePath} />
```

Comment:
What's the significance of `basename`?

Reply:
Its the url base path. This is the name used with the application reg api. I think line 75 does a good job demonstrating but I'm happy to amend it if you have suggestions.

Reply:
👍 Thanks for the explanation! I feel like `basePath` is a fairly well understood concept in Kibana. I think of it as "the app's namespace inside of the URI". If we changed `basename` to `basePath` would that carry the connotations? Just throwing it out here, feel free to disregard.